include/circt/Dialect/RTL/Combinatorial.td

```diff
@@ -163,6 +163,32 @@ def XorROp : UnaryI1ReductionRTLOp<"xorr">;
 // Other Operations
 //===----------------------------------------------------------------------===//
 
+// Extract a range of bits from the specified input.
+def StructExtractOp : RTLOp<"struct_extract", [NoSideEffect]> {
+  let summary = "Extract a named field from a struct.";
+
+  let arguments = (ins StrAttr:$field, AnyRTLStructType:$input);
+  let results = (outs AnyType:$result);
+
+  let assemblyFormat = [{
+    $field `of` $input attr-dict `:` functional-type($input, $result)
```

Comment:
I'd suggest `from` instead of `of`.

Reply:
How about "$input \`.\` $field" ?

Also, please change the syntax to just have the type of $input.  From that and the attribute we can derive the result type.

Reply:
'.' isn't a valid identifier, so I'm avoiding it to keep this compatible with declarative definitions.
But hopefully the current syntax is nicer.